backend/src/main/java/com/flightsearch/backend/Services/SearchService.java

```diff
@@ -77,6 +91,9 @@ public List<JSONObject> flightOfferSearch(String originLocationCode, String dest
     private JSONObject addAirportAndAirlinesCommonNames(String originalJson) throws IOException {
         JSONObject jsonResponse =new JSONObject(originalJson);
         JSONArray dataArray = jsonResponse.getJSONArray("data");
+        Map<String, String> deptAirportNamesByIATACode = new HashMap<>();
+        Map<String, String> arrAirportNamesByIATACode = new HashMap<>();
+        Map<String, String> airlineNamesByCarrierCode = new HashMap<>();
```

Comment:
You can improve the performance of the application by preventing the retrieval of common information, in this case the common names of airlines and airports.